lib/elixir/lib/kernel.ex

```diff
@@ -4449,9 +4449,11 @@ defmodule Kernel do
     end
   end
 
-  # TODO: Deprecate by v1.5
   @doc false
+  # TODO: Remove by 2.0
   defmacro to_char_list(arg) do
+    %{file: file, line: line} = __CALLER__
+    :elixir_errors.warn(line, file, "to_char_list/1 is deprecated, use to_charlist/1")
```

Comment:
I think Kernel.to_char_list/1 is fine as users most likely will be able to deduce it thanks to the file+line information.